gemd/entity/case_insensitive_dict.py

```diff
@@ -50,14 +55,128 @@ def get(self, key: str, default=None):
         else:
             return default
 
-    def __setitem__(self, key: str, value):
+    def __setitem__(self, key: str, value: Any) -> None:
         self._register_key(key)
         super().__setitem__(key, value)
 
-    def __contains__(self, key: str):
+    def __contains__(self, key: str) -> bool:
         return self.lowercase_dict.__contains__(key.lower())
 
-    def _register_key(self, key: str):
+    def __delitem__(self, key) -> None:
+        key = self.lowercase_dict.get(key.lower(), key)
+        super().__delitem__(key)
+        del self.lowercase_dict[key.lower()]
+
+    def clear(self) -> None:
+        """Remove all items from the dictionary."""
+        super().clear()
+        self.lowercase_dict.clear()
+
+    def pop(self, key: str, default=_RaiseKeyError) -> Any:
+        """
+        Remove and return the value for a given key from the dictionary.
+
+        If key is in the dictionary, remove it and return its value, else return default.
+        If default is not given and key is not in the dictionary, a KeyError is raised.
+
+        Parameters
+        ----------
+        key: str
+            The key to look up (possibly with a different casing).
+
+        default: Any
+            The result to return if the key is not present.
+
+        Returns
+        -------
+        Any
+            The value associated with the case-insensitive version of `key`, or None
+            if `key` is not present.
+
+        """
+        if default is _RaiseKeyError:
```

Comment:
Why not make the default `None`?

Reply:
Because `None` is a legitimate value.  I need to catch that the argument was not passed.

Reply:
Yes, I understand now